tests/modules/test_upload_contacts_from_file.py

```diff
@@ -1,237 +1,278 @@
+            parse_or_create_functional_dob("20000-20-15", "0000-20-15", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2014-35-00", "2014-35-00", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2016-12-40", "2016-12-40", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2017-02-40", "2017-02-40", 0)
+
+    def test_nonexistent_dates_for_parse_or_create_functional_dob(self):
+        with self.assertRaises(TypeError):
+            parse_or_create_functional_dob("", "", 0)
+
+        delay1 = 0
+        delay2 = 5
+        dob1 = entered_date_string_to_date("10-15-2015")
+        parsed_func_dob = parse_or_create_functional_dob("", dob1, delay1)
+        parsed_func_dob2 = parse_or_create_functional_dob("", dob1, delay2)
+        
+        self.assertEqual(parsed_func_dob, datetime(2015, 10, 15).date())
+        self.assertEqual(parsed_func_dob2, datetime(2015, 10, 20).date())
+
+    def test_real_dates_for_parse_or_create_functional_dob(self):
+        funct_dob1 = "10-15-2015"
+        funct_dob2 = "10-18-2015"
+        dob1 = entered_date_string_to_date("10-15-2015")
+        delay1 = 0
+        delay2 = 3
+        parsed_func_dob1 = parse_or_create_functional_dob(funct_dob1, dob1, delay1)
+        parsed_func_dob2 = parse_or_create_functional_dob(funct_dob2, dob1, delay2)
+        
+        self.assertEqual(parsed_func_dob1, datetime(2015, 10, 15).date())
+        self.assertEqual(parsed_func_dob2, datetime(2015, 10, 18).date())
+
+    def test_nonexistent_dates_for_entered_date_string_to_date(self):
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("")
+
+    def test_fake_dates_for_entered_date_string_to_date(self):
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("40-15-2015")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("30-15-2015")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("12-35-2015")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("2014-20-15")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("12-00-2015")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("0000-20-15")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("2014-35-00")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("2016-12-40")
+        with self.assertRaises(ValueError):
+            entered_date_string_to_date("2017-02-29")
+
+    def test_real_dates_for_entered_date_string_to_date(self):
+        date_string1 = "2014-12-20"
+        date_string2 = "2017-01-20"
+        date_string3 = "2017-01-07"
+        date_string4 = "01-15-2017"
+        date_string5 = "02-28-2017"
```

Comment:
Can you just pass these dates into the functions directly rather than assign them to meaningless intermediary variables?